### PR TITLE
Allow DynamicStruct::Reader/Builder.as<AnyStruct>

### PR DESCRIPTION
**Pull request by jparyani**
Also fix AnyStruct::Reader/Builder as methods

### CODE REVIEW

**Comment by scottpurdy** [comment]
:+1: This builds and works great (see capnproto mailing list for relevant discussion).

**Comment by kentonv** [comment]
Argh, I totally forgot this was sitting in my queue as I pushed a bugfix release just now. /facepalm

File: c++/src/capnp/any.h [hunk 1]
```diff
@@ -465,6 +467,12 @@ class AnyStruct::Builder {
 
   inline operator Reader() const { return Reader(_builder.asReader()); }
   inline Reader asReader() const { return Reader(_builder.asReader()); }
+
+  template <typename T>
+  BuilderFor<T> as() {
+    return typename T::Builder(_builder);
+  }
+  // T must be a struct type.
```
Comment (kentonv) [inline c++/src/capnp/any.h]: Unwritten KJ style rule: This comment should be inside the function block, on the first line after the '{'.

Also on line 442.

File: c++/src/capnp/any-test.c++ [hunk 2]
```diff
@@ -114,6 +114,12 @@ TEST(Any, AnyStruct) {
   checkTestMessage(root.getAnyPointerField().getAs<TestAllTypes>());
   checkTestMessage(root.asReader().getAnyPointerField().getAs<TestAllTypes>());
 
+  auto all_types = root.getAnyPointerField().getAs<AnyStruct>().as<TestAllTypes>();
```
Comment (kentonv) [inline c++/src/capnp/any-test.c++]: Unwritten KJ style rule: Use camel-case, not underscores: `allTypes`, `allTypesReader`.

**Comment by kentonv** [comment]
Code looks good, just nitpicky style comments. (Finally writing that style guide is next on my todo list!)

**Comment by jparyani** [comment]
Fixed

**Pull request merged**